nixos/modules/programs/vtune.nix

```
@@ -0,0 +1,76 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    types
+    ;
+in
+{
+  options.programs.vtune = {
+    enable = mkEnableOption "VTune Profiler";
+
+    package = mkPackageOption pkgs "intel-oneapi-vtune" {
+      nullable = true;
+      extraDescription = ''
+        Set to `null` to install only the kernel module.
+      '';
+    };
+
+    modulePackage = mkOption {
+      type = types.nullOr types.package;
+      default = config.boot.kernelPackages.vtune-sepdk.override {
+        intel-oneapi-vtune = config.programs.package or pkgs.intel-oneapi-vtune;
+      };
+      defaultText = ''
+        config.boot.kernelPackages.vtune-sepdk.override {
+          intel-oneapi-vtune = config.programs.vtune.package or pkgs.intel-oneapi-vtune;
+        }
+      '';
+      example = "config.boot.kernelPackages.vtune-sepdk";
+      description = ''
+        The package to use for the VTune Profiler kernel module.
+        Set to `null` to disable the kernel module.
+        An user should be in the `vtune` group to use the module.
+      '';
+    };
+  };
+
+  config =
+    let
+      cfg = config.programs.vtune;
+    in
+    mkIf cfg.enable (mkMerge [
+      (mkIf (cfg.package != null) { environment.systemPackages = [ cfg.package ]; })
+
+      (mkIf (cfg.modulePackage != null) {
+        boot.extraModulePackages = [ cfg.modulePackage ];
+
+        environment.systemPackages = [ cfg.modulePackage ];
+
+        users.groups.vtune.gid = config.ids.gids.vtune;
```

This group is used for accessing some devices in /dev?

---

Usually this would be enough for simple groups like this:

```suggestion
        users.groups.vtune = {};
```

---

Yes, it's for `/dev/{pax,sep5,socperf3}`.
Pushed an update.